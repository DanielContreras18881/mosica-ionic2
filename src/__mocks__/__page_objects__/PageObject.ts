export class PageObject {
  constructor(public fixture) {}

  async wait() {
    await flushPromises()
    this.fixture.detectChanges()
  }

  text() {
    return this.removeComments(this.fixture.nativeElement.innerHTML.trim())
  }

  removeComments(text) {
    return text.replace(/<!--[\s\S]*?-->/g,'')
  }

  changeInput(inputName, inputValue) {
    this.fixture.componentInstance[inputName] = inputValue
    this.fixture.detectChanges()
  }

  containsText(text) {
    expect(this.text()).toContain(text)
  }

  notContainsText(text) {
    expect(this.text()).not.toContain(text)
  }

  matchSnapshot() {
    expect(this.text()).toMatchSnapshot()
  }
}

export function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}
