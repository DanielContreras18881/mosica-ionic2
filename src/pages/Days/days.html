<ion-header>
  <ion-navbar>
    <button ion-button menuToggle>
      <ion-icon name="icon"></ion-icon>
    </button>
    <ion-title>Listaca de conciertos</ion-title>
  </ion-navbar>
</ion-header>

<ion-content>
  <ion-searchbar
    showCancelButton="false"
    (ionInput)="onSearch($event)">
  </ion-searchbar>
  <LoadSpinner [isLoading]="isLoading"></LoadSpinner>
  <ion-list>
    <div *ngFor="let day of currentDays">
      <ion-item-divider >
        {{day.day}}
      </ion-item-divider>
      <button class="gig" ion-item *ngFor="let gig of day.gigs" (click)="open($event, gig)">
        {{gig.title}} - {{gig.place}}
      </button>
    </div>
  </ion-list>
</ion-content>
