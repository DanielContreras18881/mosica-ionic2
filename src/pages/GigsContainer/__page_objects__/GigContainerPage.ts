export class GigsContainerPage {
  constructor(private fixture) {}

  async updateAsync() {
    await flushPromises();
    this.fixture.detectChanges()
  }

  text() {
    return this.fixture.nativeElement.innerHTML;
  }

  containsText(atext) {
    expect(this.text()).toContain(atext)
  }
}

export function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}
