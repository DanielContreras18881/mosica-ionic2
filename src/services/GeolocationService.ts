import {Injectable} from '@angular/core';

@Injectable()
export class GeolocationService {

  currentPosition() {
    return new Promise(function(resolve,reject){
      navigator.geolocation.getCurrentPosition(resolve,reject);
    }).then((result: any) => {
      return {latitude: result.coords.latitude, longitude: result.coords.longitude};
    });
  }

  // Example of promisify that would just work if getCurrentPosition was a node style function with callbacks
  // https://github.com/petkaantonov/bluebird/issues/1179
  // currentPositionWithPromisify() {
  //   const getCurrentPosition = promisify(navigator.geolocation.getCurrentPosition);
  //   getCurrentPosition().then((result: any) => {
  //       return {latitude: result.coords.latitude, longitude: result.coords.longitude};
  //     });
  // }
}
