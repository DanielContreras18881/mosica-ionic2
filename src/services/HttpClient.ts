import {Injectable} from '@angular/core';
import {Jsonp} from '@angular/http';
import 'rxjs/add/operator/toPromise';
import 'rxjs/add/operator/map';

@Injectable()
export class HttpClient {
  constructor(private jsonp: Jsonp) {}

  get(url) {
    return this.jsonp.request(url +'?callback=JSONP_CALLBACK')
      .map(result => {
        return { body: result.json() };
      })
      .toPromise();
  }
}
