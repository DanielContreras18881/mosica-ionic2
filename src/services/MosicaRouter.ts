import {Injectable} from '@angular/core';
import {GigPage} from '../pages/GigsDetail/GigDetail';
import {App, NavController} from 'ionic-angular';

@Injectable()
export class MosicaRouter {
  constructor(private app: App) {}

  // We should inject controller instead of App:
  // https://github.com/ionic-team/ionic/issues/9581
  navCtrl(): NavController {
    return this.app.getRootNav();
  }

  navigateToGig(gig) {
    this.navCtrl().push(GigPage, {
      gig: gig
    });
  }
}
