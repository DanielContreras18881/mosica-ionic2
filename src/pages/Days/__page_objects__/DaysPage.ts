import {PageObject} from '../../../__mocks__/__page_objects__/PageObject';

export class DaysPage extends PageObject{
  constructor(public fixture) {
    super(fixture)
  }

  clickFirstGig() {
    const button = this.fixture.nativeElement.querySelector('.gig');
    button.click();
    this.fixture.detectChanges()
  }
}

export function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}
