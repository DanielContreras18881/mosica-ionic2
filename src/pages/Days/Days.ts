import { Component } from '@angular/core';
import * as core from 'mosica-core';
import {MosicaRouter} from '../../services/MosicaRouter';

@Component({
  selector: 'page-list',
  templateUrl: 'days.html'
})
export class Days {
  currentDays: any
  isLoading: Boolean = true
  constructor(public mosicaRouter: MosicaRouter,
              public gigService: core.GigService) {


    gigService.retrieveNextGigs().then((gigsByDay) => {
      this.currentDays = gigsByDay
      this.isLoading = false
    });
  }

  open(event, gig) {
    this.mosicaRouter.navigateToGig(gig);
  }

  async onSearch(event) {
    this.currentDays = await this.gigService.searchGigsGroupedByDay(event.target.value);
  }
}
