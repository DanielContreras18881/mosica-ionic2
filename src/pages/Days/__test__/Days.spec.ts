import {TestBed, async} from '@angular/core/testing';
import {Days} from '../Days';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import * as core from 'mosica-core';
import {fakeGigsByDay} from './fake-gigs-by-day';
import {DaysPage} from '../__page_objects__/DaysPage';
import {MosicaRouter} from '../../../services/MosicaRouter';

describe('Days', () => {

  let fixture, page;
  const navigateToGigSpy = jest.fn()
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [
        NO_ERRORS_SCHEMA
      ],
      declarations: [
        Days
      ],
      providers: [
        {provide: MosicaRouter, useValue: { navigateToGig: navigateToGigSpy} },
        {provide: core.GigService, useClass: core.GigService},
      ]
    }).compileComponents();

    fixture = TestBed.createComponent(Days);
    page = new DaysPage(fixture)
  }));


  it('matches full snapshot', async() => {
    await page.updateAsync()
    page.matchSnapshot()
  });

  it('shows all gigs in first day', async() => {
    const FIRST_DAY_GIG_TEXTS = fakeGigsByDay[0].gigs.map(
      gig => gig.title + ' - ' + gig.place)
    await page.updateAsync()
    FIRST_DAY_GIG_TEXTS.map((text) =>
      page.containsText(text)
    )
  });

  it('navigates to first gig detail', async () => {
    const FIRST_GIG = fakeGigsByDay[0].gigs[0]
    await page.updateAsync()
    page.clickFirstGig();
    expect(navigateToGigSpy).toHaveBeenCalledWith(FIRST_GIG);
  });

  //
  // it('navigates to second gig detail', async () => {
  //   const SECOND_GIG_ID = '2222222'
  //   page.clickSecondGig();
  //   expect(navigateToGigMock).toHaveBeenCalledWith(SECOND_GIG_ID);
  // });

});
