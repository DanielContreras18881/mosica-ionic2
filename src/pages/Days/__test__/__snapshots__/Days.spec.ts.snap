// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Days matches full snapshot 1`] = `
"<ion-header>
  <ion-navbar>
    <button ion-button=\\"\\" menutoggle=\\"\\">
      <ion-icon name=\\"icon\\"></ion-icon>
    </button>
    <ion-title>Listaca de conciertos</ion-title>
  </ion-navbar>
</ion-header>

<ion-content>
  <ion-searchbar showcancelbutton=\\"false\\">
  </ion-searchbar>
  <loadspinner></loadspinner>
  <ion-list>
    <div>
      <ion-item-divider>
        Lunes, 18 de Septiembre
      </ion-item-divider>
      <button class=\\"gig\\" ion-item=\\"\\">
        Anarchy in the JS - SCBCN
      </button><button class=\\"gig\\" ion-item=\\"\\">
        Vegan Microservices - Around the world
      </button>
    </div>
  </ion-list>
</ion-content>"
`;
